_posts/2023-08-23-cluster-metrics-monitors-blog.md

```diff
@@ -0,0 +1,168 @@
+
+## Monitoring cluster CPU utilization
+
+By creating a cluster metrics monitor that calls the `_cluster/stats` API, you can receive alerts when the CPU utilization of the cluster reaches a certain threshold. To generate an alert when the CPU utilization reaches or exceeds 60%, use the following steps:
+
+1. From the top menu, select **OpenSearch Plugins** > **Alerting**.
+1. In the **Monitors** tab, select **Create monitor**.
+1. Select the **Per cluster metrics monitor** option. For demonstration purposes, you’ll configure this monitor to run every minute by leaving the default values under **Schedule**.
+1. Under **Query** > **Request type**, select **Cluster stats**. In the response preview, you can see that the current CPU utilization for this cluster is 0%.
+    ![Configuring the request type](/assets/media/blog-images/2023-08-23-cluster-metrics-monitors-blog/cluster-metrics-v29-cpu2-request-type.png){: .img-fluid }
+1. Configure a trigger condition that compares the CPU percent to your desired threshold: 
+    ```bash
+    ctx.results[0].nodes.process.cpu.percent >= 60
+    ```
+    In this example, you can see that the trigger condition response is `false` because the CPU utilization for the cluster is currently `0%`.
+    ![Configuring a trigger](/assets/media/blog-images/2023-08-23-cluster-metrics-monitors-blog/cluster-metrics-v29-cpu3-trigger.png){: .img-fluid }
+1. Configure notification actions as desired. This example uses a dummy channel for demonstration purposes. 
+    ![Configuring notifications](/assets/media/blog-images/2023-08-23-cluster-metrics-monitors-blog/cluster-metrics-v29-cpu4-notification.png){: .img-fluid }
+    As you can see in the example, you can include `ctx.results.0.nodes.process.cpu.percent` to print the CPU utilization in the notification message:
+   
+    ```plaintext
+    The current CPU usage for the cluster is {% raw %}{{ctx.results.0.nodes.process.cpu.percent}}%{% endraw %}`.
+
+    Monitor {% raw %}{{ctx.monitor.name}}{% endraw %} just entered alert status. Please investigate the issue.
+    - Trigger: {% raw %}{{ctx.trigger.name}}{% endraw %}
+    - Severity: {% raw %}{{ctx.trigger.severity}}{% endraw %}
+    - Period start: {% raw %}{{ctx.periodStart}}{% endraw %}
+    - Period end: {% raw %}{{ctx.periodEnd}}{% endraw %}
+    ```
+1. Select **Create** to create your monitor!
+
+## Monitoring cluster JVM memory pressure
+
+You can also use the `_cluster/stats` API to receive alerts when the JVM memory pressure of the cluster reaches a certain threshold. Let’s say you want to generate an alert when the JVM memory pressure reaches or exceeds 75%. This example requires a little more understanding of trigger conditions to configure, because the `_cluster/stats` API doesn’t return the percentage of the heap that’s used; it returns `heap_used_in_bytes` and `heap_max_in_bytes`.
```

Comment:
```suggestion
You can also use the `_cluster/stats` API to receive alerts when the JVM memory pressure of the cluster reaches a certain threshold. Let’s say you want to generate an alert when the JVM memory pressure reaches or exceeds 75%. This example requires a little more understanding of trigger conditions to configure because the `_cluster/stats` API doesn’t return the percentage of the heap that’s used; it returns `heap_used_in_bytes` and `heap_max_in_bytes`.
```